plenum/test/view_change/test_primary_selector.py

```diff
@@ -0,0 +1,169 @@
+from typing import Optional
+
+import base58
+import pytest
+from plenum.server.primary_selector import PrimarySelector
+from plenum.common.types import ViewChangeDone
+from plenum.server.replica import Replica
+from plenum.common.util import get_strong_quorum
+from plenum.common.ledger_manager import LedgerManager
+from plenum.common.ledger_manager import Ledger
+
+
+whitelist = ['but majority declared']
+
+
+class FakeLedger():
+    def __init__(self, ledger_id, size):
+        self._size = size
+        self.root_hash = (base58.alphabet[ledger_id] * 43)
+        self.hasher = None
+
+    def __len__(self):
+        return self._size
+
+
+# Question: Why doesn't this subclass Node.
+class FakeNode():
+    def __init__(self):
+        self.name = 'Node1'
+        self.f = 1
+        self.replicas = []
+        self.viewNo = 0
+        self.rank = None
+        self.allNodeNames = [self.name, 'Node2', 'Node3', 'Node4']
+        self.totalNodes = len(self.allNodeNames)
+        self.ledger_ids = [0]
+        self.replicas = [
+            Replica(node=self, instId=0, isMaster=True),
+            Replica(node=self, instId=1, isMaster=False),
+            Replica(node=self, instId=2, isMaster=False),
+        ]
+        self._found = False
+        self.ledgerManager = LedgerManager(self, ownedByNode=True)
+        ledger0 = FakeLedger(0, 10)
+        ledger1 = FakeLedger(1, 5)
+        self.ledgerManager.addLedger(0, ledger0)
+        self.ledgerManager.addLedger(1, ledger1)
+
+    def get_name_by_rank(self, name):
+        # This is used only for getting name of next primary, so
+        # it just returns a constant
+        return 'Node2'
+
+    def primary_found(self):
+        self._found = True
+
+    def is_primary_found(self):
+        return self._found
+
+    @property
+    def master_primary_name(self) -> Optional[str]:
+        nm = self.replicas[0].primaryName
+        if nm:
+            return Replica.getNodeName(nm)
+
+
+def testHasViewChangeQuorum():
+    """
+    Checks method _hasViewChangeQuorum of SimpleSelector 
+    """
+
+    instance_id = 0
+    last_ordered_seq_no = 0
+    selector = PrimarySelector(FakeNode())
```

Comment:
If we get rid of node dependency in PrimarySelector, then we would not need FakeNode implementation here